packages/@okta/vuepress-site/docs/reference/architecture-center/directory-coexistence/lab-tutorial-3-generic-database/index.md

```diff
@@ -0,0 +1,316 @@
+
+   </div>
+
+   > **Note:** Okta automatically manages a group called **Everyone**, which includes all users. The **Everyone** group also includes all applications. You may see different applications assigned to each user if you have completed other examples in this tutorial.
+
+   You can select other tabs for additional information about each user.
+
+## Update the application to use Universal Directory
+
+Now the users have been mirrored in Universal Directory, you can reconfigure the application to use Universal Directory as its identity provider.
+
+1. In the `okta-reference-coexistence-db-example` directory, stop the current running containers with <kbd>Ctrl</kbd> + <kbd>C</kbd>.
+
+1. Remove the containers:
+
+   ```bash
+   docker compose down
+   ```
+
+1. Create an Okta OpenID Connect Application and register the application with Okta:
+
+   ```bash
+   okta start
+   ```
+
+   The `okta start` CLI command creates and registers an OAuth 2.0/OpenID Connect client with Okta. Okta considers this an OpenID Connect application. When this OpenID Connect application is registered, the `ISSUER`, `CLIENT_ID` and `CLIENT_SECRET` variables in the `okta-reference-coexistence-db-example/.env` file are updated for authentication with Okta, similar to:
+
+   ```js
+   ISSUER=https://dev-133337.okta.com/oauth2/default
+   CLIENT_ID=0oa6iartjn5iuoKzr5d7
+   CLIENT_SECRET=fF4BMQ0OhGp3D63s71BBTZoztIWqICdrCXBdkCtn
+   ```
+
+   The sample web application in the Docker container uses the client ID and client secret to communicate with the Okta OpenID Connect application to use Okta as the identity provider. See [Values and Variables](/docs/reference/architecture-center/directory-coexistence/lab-prerequisites/#values-and-variables).
+
+1. Rebuild the application:
+
+   ```bash
+   docker compose build
+   ```
+
+1. Start the application:
+
+   ```bash
+   docker compose up
+   ```
+
+1. Open a private/incognito window and navigate to the same URL as before, `http://localhost:8080`. This time you are redirected to Okta to sign-in. Sign in with the user's email address and password: `user1@example.com` and `password`.
+
+   <div class="half border">
+
+   ![Sign-in dialog including username and password fields](/img/architecture/directory-coexistence/db-to-okta-sign-in-to-db-example.png)
+
+   </div>
+
+   When the user is authenticated by Okta, you will see the following displayed:
+
+   <div class="three-quarter border">
+
+   ![Successful sign-in response](/img/architecture/directory-coexistence/db-to-okta-successful-signin.png)
+
+   </div>
+
+## Account for hash types Okta does not support
+
+You can use the [Okta Password Inline Import Hook](/docs/reference/password-hook/) to import user records with an [unsupported password hash type](/docs/reference/api/users/#hashed-password-object) without needing to reset the user's password. The first time the user with an unsupported hash tries to sign in to your application, the following flow occurs:
```

Comment:
A couple of other Password Import Inline Hook references you may want to reference, see this Guide: https://developer.okta.com/docs/guides/password-import-inline-hook/nodejs/main/

And this blog post: https://developer.okta.com/blog/2020/09/18/password-hook-migration

Reply:
Great links. Will add them to the resources page for the second iteration of the paper. Thanks